### DISCUSSION TITLE
ruby 3.0.4 is deprecated, can we have it updated or at least can we more than 1 option for ruby version

### DISCUSSION THREAD

**Discussion by wei-addepar** [codemagic-ci-cd/codemagic-docs#2777] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

iOS (Swift, Objective-C)

### Steps to reproduce

1. mac_mini_m2 default to ruby 3.0.4, which is deprecated
2. trying downgrade my local ruby to 3.0.4 was not successful
3. there's only 1 option when it comes to ruby versions
4. please include more recent versions

### Expected results

More recent ruby version options provided so that code magic can match what I have locally.

### Actual results

Either the build process fails because of my bundler version didn't match or complains about I should use bundler to execute `pod install`

**Comment by icarusdust** [comment, marked as answer]
Hi @wei-addepar, I replied to your query in our private thread.

**Comment by aneletechtides** [comment]
I'm having the same issue, any update?

**Reply by wei-addepar** [reply]
they said
> I will share it with the team and get back to you on it. 

So I guess no, yet

**Reply by icarusdust** [reply]
Hi @aneletechtides, you can specify a ruby version in `codemagic.yaml `:

```
workflows:
  default-workflow:
    name: Default Workflow
    environment:
       vars:
       ruby: REQUIRED_VERSION_HERE
```
In the meantime, we are internally discussing adding support to newer Ruby versions.

**Reply by icarusdust** [reply]
You can also try installing the required Ruby version during build by adding the following script at the top of the scripts section in `codemagic.yaml` or in a post-clone script in the Workflow Editor:

```
brew upgrade ruby-build
rbenv install 3.3.4
echo "RBENV_VERSION=3.3.4" >> $CM_ENV
```

**Comment by wei-addepar** [comment]
Hi Nihal,
Thanks for replying.
Installing a new ruby version takes a long time. That's the main reason
that I don't want to do that.

On Fri, Nov 8, 2024 at 8:48 AM Nihal Agazade ***@***.***>
wrote:

> You can also try installing the required Ruby version during build by
> adding the following script at the top of the scripts section in
> codemagic.yaml or in a post-clone script in the Workflow Editor:
>
> brew upgrade ruby-build
> rbenv install 3.3.4
> echo "RBENV_VERSION=3.3.4" >> $CM_ENV
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/codemagic-ci-cd/discussions/2777#discussioncomment-11189386>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AWWUHV6IKYYQMQRK3NOF74LZ7S6JVAVCNFSM6AAAAABPVAB2FCVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCMJYHEZTQNQ>
> .
> You are receiving this because you were mentioned.Message ID:
> <codemagic-ci-cd/codemagic-docs/repo-discussions/2777/comments/11189386@
> github.com>
>


-- 
*Wei Zhang *
*Senior Software Engineer*
***@***.***

**Reply by icarusdust** [reply]
Hi Wei, it is so your pipeline is not blocked while we are checking to update the pre-installed ruby versions
